src/AppBundle/Entity/Component.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+    namespace AppBundle\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+    use Symfony\Component\Validator\Constraints as Assert;
+
+    /**
+    * Component
+    *
+    * @ORM\Table(name="sqldashboard_component")
+    * @ORM\Entity(repositoryClass="AppBundle\Repository\ComponentRepository")
+    */
+    class Component
+    {
+        /**
+        * @var int
+        *
+        * @ORM\Column(name="id", type="integer")
+        * @ORM\Id
+        * @ORM\GeneratedValue(strategy="AUTO")
+        */
+        private $id;
+
+        /**
+        * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Dashboard")
+        * @ORM\JoinColumn(nullable=false)
+        */
+        private $dashboard;
+
+        /**
+        * @var string
+        *
+        * @ORM\Column(name="name", type="string", length=255)
+        *
+        * @Assert\NotBlank()
+        * @Assert\Length(
+        *      max = 255,
+        *      maxMessage = "The name cannot be longer than {{ limit }} characters"
+        * )
+        *
+        */
+        private $name;
+
+        /**
+        * @var string
+        *
+        * @ORM\Column(name="requestSQL", type="text")
+        *
+        * @Assert\NotBlank()
+        *
+        */
+        private $requestSQL;
+
+        /**
+        * @var string
+        *
+        * @ORM\Column(name="typeGraph", type="string", length=255)
+        */
```

Comment:
It does not appear in the form, it's fixed by the action of a button I still have to assign assert to it?